internal/services/v1/permissions.go

```diff
@@ -535,86 +536,157 @@ func (ps *permissionServer) LookupSubjects(req *v1.LookupSubjectsRequest, resp v
+					if err != nil {
+						return fmt.Errorf("error when resolving excluded subject: %w", err)
+					}
+
+					if resolvedExcludedSubject == nil {
+						continue
+					}
+
+					excludedSubjects = append(excludedSubjects, resolvedExcludedSubject)
+				}
+
+				subject, err := foundSubjectToResolvedSubject(ctx, foundSubject, caveatContext, ds)
+				if err != nil {
+					return fmt.Errorf("error when resolving subject: %w", err)
+				}
+				if subject == nil {
+					continue
+				}
+
+				// NOTE: we need to recompute the cursor here because we get multiple results back from DispatchLookupSubjects
+				// in one message.
+				dispatchCursor, err := graph.CursorForFoundSubjectID(subject.SubjectObjectId, result.AfterResponseCursor)
 				if err != nil {
 					return err
 				}
 
-				if resolvedExcludedSubject == nil {
-					continue
+				// Update the existing internal cursor for encoding.
+				internalResponseCursor.GetV1().DispatchVersion = dispatchCursor.DispatchVersion
+				internalResponseCursor.GetV1().Sections = dispatchCursor.Sections
+
+				encodedCursor, err := cursor.Encode(internalResponseCursor)
+				if err != nil {
+					return err
 				}
 
-				excludedSubjects = append(excludedSubjects, resolvedExcludedSubject)
-			}
+				currentCursor = dispatchCursor
 
-			subject, err := foundSubjectToResolvedSubject(ctx, foundSubject, caveatContext, ds)
-			if err != nil {
-				return err
-			}
-			if subject == nil {
-				continue
-			}
+				if subject.SubjectObjectId != tuple.PublicWildcard {
+					countSubjectsFound++
+					if req.OptionalConcreteLimit > 0 && remainingConcreteLimit <= 0 {
+						return nil
+					}
+					remainingConcreteLimit--
+				}
 
-			err = resp.Send(&v1.LookupSubjectsResponse{
-				Subject:            subject,
-				ExcludedSubjects:   excludedSubjects,
-				LookedUpAt:         revisionReadAt,
-				SubjectObjectId:    foundSubject.SubjectId,    // Deprecated
-				ExcludedSubjectIds: excludedSubjectIDs,        // Deprecated
-				Permissionship:     subject.Permissionship,    // Deprecated
-				PartialCaveatInfo:  subject.PartialCaveatInfo, // Deprecated
-			})
-			if err != nil {
-				return err
+				err = resp.Send(&v1.LookupSubjectsResponse{
+					Subject:            subject,
+					ExcludedSubjects:   excludedSubjects,
+					LookedUpAt:         revisionReadAt,
+					SubjectObjectId:    foundSubject.SubjectId,    // Deprecated
+					ExcludedSubjectIds: excludedSubjectIDs,        // Deprecated
+					Permissionship:     subject.Permissionship,    // Deprecated
+					PartialCaveatInfo:  subject.PartialCaveatInfo, // Deprecated
+					AfterResultCursor:  encodedCursor,
+				})
+				if err != nil {
+					return err
+				}
 			}
-		}
-
-		dispatchpkg.AddResponseMetadata(respMetadata, result.Metadata)
-		return nil
-	})
 
-	bf, err := dispatch.NewTraversalBloomFilter(uint(ps.config.MaximumAPIDepth))
-	if err != nil {
-		return err
-	}
+			dispatchpkg.AddResponseMetadata(respMetadata, result.Metadata)
+			return nil
+		})
 
```

Comment:
Was it removed because we are not single-flighting `LookupSubject` dispatches?

Reply:
This seems to be a rebase error... I've added it back in, but we should add a test to catch this as a followup